### DISCUSSION TITLE
Why is `just` slower with default target?

### DISCUSSION THREAD

**Discussion by bukowa** [casey/just#2187] [Q&A]
Hey I wonder why is one slower than the other?
This is with this short `justfile` - the more recipes I have the slower it is.

```justfile
# default target
_:
    @just --list
```

```bash
$ time just
Available recipes:

real    0m0.044s
user    0m0.000s
sys     0m0.000s

````
```bash
$ time just --list
Available recipes:

real    0m0.017s
user    0m0.000s
sys     0m0.000s

````

**Comment by bukowa** [comment]
https://github.com/casey/just/issues/2189

**Comment by casey** [comment]
This:

```
_:
    @just --list
```

Causes just to invoke itself as a subprocess, whereas `just --list` doesn't need to invoke a subprocess, so the slowdown is expected.
